deepchem/feat/molecule_featurizers/dmpnn_featurizer.py

```diff
@@ -254,3 +261,226 @@ def map_reac_to_prod(
                   List[int]] = (reac_id_to_prod_id, only_prod_ids,
                                 only_reac_ids)
   return mappings
+
+
+class DMPNNFeaturizer(MolecularFeaturizer):
+  """This class is a featurizer for Directed Message Passing Neural Network (D-MPNN) implementation
+
+  The default node(atom) and edge(bond) representations are based on
+  `Analyzing Learned Molecular Representations for Property Prediction paper <https://arxiv.org/pdf/1904.01561.pdf>`_.
+
+  The default node representation are constructed by concatenating the following values,
+  and the feature length is 133.
+
+  - Atomic num: A one-hot vector of this atom, in a range of first 100 atoms.
+  - Degree: A one-hot vector of the degree (0-5) of this atom.
+  - Formal charge: Integer electronic charge, -1, -2, 1, 2, 0.
+  - Chirality: A one-hot vector of the chirality tag (0-3) of this atom.
+  - Number of Hydrogens: A one-hot vector of the number of hydrogens (0-4) that this atom connected.
+  - Hybridization: A one-hot vector of "SP", "SP2", "SP3", "SP3D", "SP3D2".
+  - Aromatic: A one-hot vector of whether the atom belongs to an aromatic ring.
+  - Mass: Atomic mass * 0.01
+
+  The default edge representation are constructed by concatenating the following values,
+  and the feature length is 14.
+
+  - Bond type: A one-hot vector of the bond type, "single", "double", "triple", or "aromatic".
+  - Same ring: A one-hot vector of whether the atoms in the pair are in the same ring.
+  - Conjugated: A one-hot vector of whether this bond is conjugated or not.
+  - Stereo: A one-hot vector of the stereo configuration (0-5) of a bond.
+
+  If you want to know more details about features, please check the paper [1]_ and
+  utilities in deepchem.utils.molecule_feature_utils.py.
+
+  Examples
+  --------
+  >>> smiles = ["C1=CC=CN=C1", "C1CCC1"]
+  >>> featurizer = DMPNNFeaturizer()
+  >>> out = featurizer.featurize(smiles)
+  >>> type(out[0])
+  <class 'deepchem.feat.graph_data.GraphData'>
+  >>> out[0].num_node_features
+  133
+  >>> out[0].node_features.shape
+  (6, 133)
+  >>> out[0].node_features_zero_padded.shape
+  (7, 133)
+  >>> out[0].num_edges
+  12
+  >>> out[0].concatenated_features_zero_padded.shape
+  (13, 147)
+  >>> len(out[0].mapping)
+  13
+
+  References
+  ----------
+  .. [1] Kearnes, Steven, et al. "Molecular graph convolutions: moving beyond fingerprints."
+     Journal of computer-aided molecular design 30.8 (2016):595-608.
+
+  Note
+  ----
+  This class requires RDKit to be installed.
+  """
+
+  def __init__(self,
+               features_generator=None,
+               is_adding_hs: bool = False,
+               use_original_atom_ranks: bool = False,
+               features_scaling: bool = False,
+               atom_descriptor_scaling: bool = False,
+               bond_feature_scaling: bool = False) -> None:
```

Comment:
`__init__` doesn't need a return type annotation

Reply:
ok noted. I will make the required changes